File: core/src/main/java/com/pluxity/facility/entity/Building.java
package com.pluxity.facility.entity;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;

@Entity
@Table(name = "building")
@DiscriminatorValue("BUILDING")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@ConditionalOnProperty(name = "facility.building.enabled", havingValue = "true")
public class Building extends Facility {
    @Builder
    public Building(String name, String description) {
        super(name, description);
    }
}

File: core/src/main/kotlin/com/pluxity/building/Building.kt
package com.pluxity.building

import com.pluxity.facility.Facility
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity
@Table(name = "building")
@DiscriminatorValue("BUILDING")
class Building(
    name: String,
    description: String?,
) : Facility(name = name, description = description)
